### DISCUSSION TITLE
ERR_MODULE_NOT_FOUND ... node_modules/@maxgraph/core/lib/view/mixins/_graph-mixins-types'

### DISCUSSION THREAD

**Discussion by sdgoldstein** [maxGraph/maxGraph#597] [Q&A] [answered]
Apologies, as I'm guessing this is something in my config, but I can't seem to find it.  Any help would be appreciated.

I have two apps using MaxGraph - 1 working and 1 not working. Both are running in a sveltekit app.  In the app not working, I've checked tsconfig, vite.config.ts, and svelte.config.js.  "moduleResolution" is set to "bundler" in both apps.  The error is below.  Any suggestions?

Thanks!

```
Error [ERR_MODULE_NOT_FOUND]: Cannot find module ... '/node_modules/@maxgraph/core/lib/view/mixins/_graph-mixins-types' imported from ... /node_modules/@maxgraph/core/lib/index.js
    at new NodeError (node:internal/errors:399:5)
    at finalizeResolution (node:internal/modules/esm/resolve:231:11)
    at moduleResolve (node:internal/modules/esm/resolve:850:10)
    at defaultResolve (node:internal/modules/esm/resolve:1058:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:835:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:416:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:76:40)
    at link (node:internal/modules/esm/module_job:75:36)
```

**Comment by sdgoldstein** [comment]
I tried starting over, creating a brand new svelte app, and ran into the same problem:

1.  Create new app:

```
npx sv create my-app
cd my-app
npm install
npm run dev
```

2.  Verified it worked
3.  Added maxgraph dependency

`npm i @maxgraph/core `  

4.  Added import to the default route, +page.svelte:

```
<script>
    import { Graph } from "@maxgraph/core";
</script>
```
5.  Refreshed the page

**Result** - Same error above

**Comment by HalilFocic** [comment, marked as answer]
Hi @sdgoldstein , 

I never did Svelte so it took me a bit of time to figure out what was the issue. In SvelteKit there is code that runs on the server and there is code that runs on the client. Our package is meant to run inside browser since its dependant on browser APIs to work.

I've managed to change the code and importing a bit and render the graph from the example we give in our README.md file on github:
![image](https://github.com/user-attachments/assets/ec09f65c-a625-4308-9ce8-27b27b381995)

Here is the code I used to achieve this inside page.svelte file:
```
//+page.svelte

<h1>Welcome to SvelteKit</h1>
<p>Visit <a href="https://svelte.dev/docs/kit">svelte.dev/docs/kit</a> to read the documentation</p>
<div id="graph-container"></div>

<script lang="ts">
  import { browser } from '$app/environment';
  import { onMount } from 'svelte';

  onMount(async () => {
      if (browser) {
          const MaxGraph = await import('@maxgraph/core');
          const container = document.getElementById('graph-container');
          const graph = new MaxGraph.Graph(container);
          console.log(graph);
          const parent = graph.getDefaultParent();
          graph.batchUpdate(()=>{
              const vertex01 = graph.insertVertex({
                  parent,
                  position: [10, 10],
                  size: [100, 100],
                  value: 'rectangle',
              });
              const vertex02 = graph.insertVertex({
                  parent,
                  position: [350, 90],
                  size: [50, 50],
                  style: {
                      fillColor: 'orange',
                      shape: 'ellipse',
                      verticalAlign: 'top',
                      verticalLabelPosition: 'bottom',
                  },
                  value: 'ellipse',
              });
              graph.insertEdge({
                  parent,
                  source: vertex01,
                  target: vertex02,
                  value: 'edge',
                  style: {
                      edgeStyle: 'orthogonalEdgeStyle',
                      rounded: true,
                  },
              });
          })
      }
  });
</script>
```

Hope this was able to make your app running, I am not really sure where on SvelteKit docs do they explain this feature but basically your SvelteKit is node server that is listening for requests. In order to optimize the amount of data you receive, it will try to render the components on the server and respond with HTML and as minimal javascript as possible. This breaks our graph since you need to make sure that your code runs in browser only, and not the server.

**Comment by sdgoldstein** [comment]
Of course!  Thank you!  Definitely a bit of a face/palm moment.  I knew that Sveltekit does both SSR and CSR, but I clearly forgot how it worked and thought it was trying to load and fail in the browser when it wasn't.

FWIW - there is an option to force client side rendering ([page options](https://svelte.dev/docs/kit/page-options)), which enables not having to do the dynamic import as you've done above.

Thanks, again!

**Reply by tbouffard** [reply]
Hi, and great to here that your problem is solved.

Be aware that an sveltekit example exists here: https://github.com/maxGraph/maxgraph-integration-examples/tree/main/projects/sveltekit-ts
Apparently, it disables SSR in https://github.com/maxGraph/maxgraph-integration-examples/blob/main/projects/sveltekit-ts/src/routes/%2Bpage.ts
